website/docs/components/rich-tooltip/partials/code/component-api.md

```diff
@@ -0,0 +1,103 @@
+  </C.Property>
+  <C.Property @name="toggleAriaLabel" @type="string">
+    The `aria-label` value applied to the HTML element that acts as "toggle".
+  </C.Property>
+  <C.Property @name="popoverPlacement" @type="enum" @values={{array "top" "top-start" "top-end" "right" "right-start" "right-end" "bottom" "bottom-start" "bottom-end" "left" "left-start" "left-end"}} @default="bottom">
+    Placement for the preferred starting position of the tooltip relative to the toggle element.
+    <br />
+    <br />
+    _Notice: if `@enableCollision` is set, the popover will automatically shift position to remain visible when near the edges of the screen regardless of the starting placement._
+  </C.Property>
+  <C.Property @name="popoverOffset" @type="number|object" @default="12">
+    A number that represents the “distance” between the tooltip and the toggle element (used to control the relative position of the arrow in relation to the toggle). An object can also be passed, which enables to individually configure different axis.
+    <br />
+    <br />
+    For details see: [Floating UI > Offset > Options](https://floating-ui.com/docs/offset#options).
+  </C.Property>
+  <C.Property @name="popoverWidth" @type="string" @valueNote="any valid CSS width (px, rem, etc)">
+    By default, the tooltip adapts to its content size. If a `@popoverWidth` parameter is provided then the tooltip will have a fixed width.
+  </C.Property>
+  <C.Property @name="popoverHeight" @type="string" @valueNote="any valid CSS height (px, rem, etc)">
+    By default, the tooltip adapts to its content size. If a `@popoverHeight` parameter is provided then the tooltip will have a fixed height.
+  </C.Property>
+  <C.Property @name="enableSoftEvents" @type="boolean" @default="true" @values={{array "true" "false"}}>
+    Assigns "soft" events listeners (`mouseEnter/Leave` + `focusIn/Out`) to the toggle, to control the visibility of the tooltip.
+  </C.Property>
+  <C.Property @name="enableClickEvents" @type="boolean" @default="false" @values={{array "true" "false"}}>
+    Assigns a "click" event listener (`onClick`) to the toggle, to control the visibility of the tooltip.
+    <br />
+    <br />
+    _Notice: enabling "click" event listeners will automatically disable "soft" event listeners._
+  </C.Property>
+  <C.Property @name="enableCollisionDetection" @type="boolean|string" @values={{array "true" "false" "flip" "shift" "auto"}} @default="false">
+    This property controls if the tooltip should automatically adapt its position to remain visible when near the edges of the screen. It can be enabled for both axes setting it to `true`, or a single axes can be chosen passing the `flip` or `shift` value. If set to `auto` it will automatically place the popover in the position where there's more space available, but in this case it will ignore the `popoverPlacement` value.
```

Comment:
Is "viewport" more appropriate here?

Maybe also consider adding something that would make `flip` and `shift` more understandable. If a consumer isn't familiar with FloatingUI, I would imagine using X and Y axes would be easier to understand. Something like:

...by passing the `flip` (Y axis) or `shift` (X axis) value.

```suggestion
    This property controls if the tooltip should automatically adapt its position to remain visible when near the edges of the viewport. It can be enabled for both axes by setting it to `true`, or a single axis can be chosen by passing the `flip` or `shift` value. If set to `auto` it will automatically place the popover in the position where there's more space available, but in this case it will ignore the `popoverPlacement` value.
```

Reply:
From what I understand, the `flip/shift` are in relation to the axes of alignment, which are determined by the "placement": 
- https://floating-ui.com/docs/shift#mainaxis
- https://floating-ui.com/docs/flip#mainaxis

so they're not necessarily X/Y axis. Suggestions on how we could make this more approachable (considered that this should be an edge case in which consumers are trying to customize a complex behaviour)?